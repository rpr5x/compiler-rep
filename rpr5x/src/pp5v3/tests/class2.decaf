class Point {
  int x;
  int y;
  void Init(int x, int y)
  {
     this.x = x;
      this.y = y;
  }
  void PrintSelf()
  {
     Print("[",x,", ",y,"]\n");
  }
  void PrintBoth(Point other)
  {
     PrintSelf();
     other.PrintSelf();
     Print(x, other.x);
  }


  bool equals(Point other) {
    return (x == other.x && y == other.y);
}

}

void main() {
  Point p;
  Point q;

  p = New(Point);
  p.Init(3, 5);
  p.PrintSelf();
  q = New(Point);
  q.Init(5, 6);
  q.PrintSelf();
  p.PrintBoth(q);
  Print(p.equals(q));
  Print(q.equals(q));
}