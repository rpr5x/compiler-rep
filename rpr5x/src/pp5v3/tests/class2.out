Loaded: /if14/ww6r/WorkingDisk/TA/compiler/exceptions/exceptions.s
[3, 5]
[5, 6]
[3, 5]
[5, 6]
35falsetrue